backend/src/web_scraper.ml

```diff
@@ -29,3 +29,850 @@ module Curl = struct
 end
 
 let fetch_exn ~url = Curl.get_exn ~url
+
```

Comment:
As discussed in-person, deterministic tests are nice. Mocking out the "hitting the network" things is good.